app/models/student.rb

```diff
@@ -78,6 +80,15 @@ def absences
     cohort.number_of_days_since_start - attendance_records.count
   end
 
+  def find_rating(internship)
+    ratings.where(internship_id: internship.id).first
+  end
+
+  def self.find_students_by_interest(internship, interest_level)
+    students = Student.all.drop_while { |student| !student.internships.include?(internship) }
+    students.select { |student| student.find_rating(internship).interest == interest_level }
+  end
```

Comment:
this should be refactored to use Active Record. right now, it pulls all of the students, and all of their internships out, and then filters them in memory. this is really inefficient; we want the db to do the work. feel free to punt on this if you aren't comfortable with it, but open a Github issue so that we don't lose track.